qfieldsync/gui/project_configuration_widget.py

```diff
@@ -229,21 +269,56 @@ def apply(self):
         self.cloudLayersConfigWidget.apply()
         self.cableLayersConfigWidget.apply()
 
+        # Base map settings
         self.__project_configuration.create_base_map = (
             self.createBaseMapGroupBox.isChecked()
         )
+
+        if self.singleLayerRadioButton.isChecked():
+            self.__project_configuration.base_map_type = (
+                ProjectProperties.BaseMapType.SINGLE_LAYER
+            )
+        else:
+            self.__project_configuration.base_map_type = (
+                ProjectProperties.BaseMapType.MAP_THEME
+            )
+
         self.__project_configuration.base_map_theme = (
             self.mapThemeComboBox.currentText()
         )
 
-        # try/pass these because the save button is global for all
+        # try/pass layer ID fetching because the save button is global for all
         # project settings, not only QField
         try:
             self.__project_configuration.base_map_layer = (
                 self.layerComboBox.currentLayer().id()
             )
         except AttributeError:
             pass
+
+        # Geofencing settings
+        self.__project_configuration.geofencing_is_active = (
+            self.geofencingGroupBox.isChecked()
+        )
+
+        try:
```

Comment:
Do you have a concrete example when we need the `try/except`? I see it is also 10 lines above, but on this line it seems you handle this with the `if/else` check for `currentLayer()` existance.